src/index.js

```diff
@@ -36,6 +36,11 @@ class SmartChart extends React.Component {
             this.mainStore.chart.onSymbolChange = onSymbolChange;
         }
     }
+    componentWillMount() {
+        if ( this.mainStore && this.mainStore.chart) {
+            this.mainStore.chart.setIsMobile(this.props.isMobile);
```

Comment:
I don't think we need this `setIsMobile` method. Just set `isMobile` directly. If `isMobile` is `undefined` it will be false anyway.

Reply:
@brucebinary  I am afraid if `this.mainStore.chart` get `undefined`

Reply:
@shayanbinary hm. True. But you are only setting `isMobile` in one place. What you can do is change the condition to 

`this.mainStore && this.mainStore.chart && this.props.isMobile !== undefined`

and we won't need `setIsMobile`

Reply:
actually why would `this.mainStore` and `this.mainStore.chart` be `undefined`? The condition should really just be `this.props.isMobile !== undefined`

Reply:
In fact you shouldn't need to override `componentWillMount`. Just add

`this.mainStore.chart.isMobile = this.props.isMobile || false;`

to `componentDidMount`.